test/integration/functional_test_tunnel_test.go

```diff
@@ -131,6 +133,69 @@ func validateTunnelStart(ctx context.Context, t *testing.T, profile string) {
 	tunnelSession = *ss
 }
 
+// validateNoSecondTunnel ensures only 1 tunnel can run simultaneously
+func validateNoSecondTunnel(ctx context.Context, t *testing.T, profile string) {
+	checkRoutePassword(t)
+
+	type SessInfo struct {
+		Stdout   string
+		Stderr   string
+		ExitCode int
+	}
+
+	sessCh := make(chan SessInfo)
+	sessions := make([]*StartSession, 2)
+
+	var runTunnel = func(idx int) {
+		args := []string{"-p", profile, "tunnel", "--alsologtostderr"}
+
+		ctx2, cancel := context.WithTimeout(ctx, Seconds(15))
+		defer cancel()
+		session, err := Start(t, exec.CommandContext(ctx2, Target(), args...))
+		if err != nil {
+			t.Errorf("failed to start tunnel: %v", err)
+		}
+		sessions[idx] = session
+
+		stderr, err := io.ReadAll(session.Stderr)
+		if err != nil {
+			t.Logf("Failed to read stderr: %v", err)
+		}
+		stdout, err := io.ReadAll(session.Stdout)
+		if err != nil {
+			t.Logf("Failed to read stdout: %v", err)
+		}
+
+		exitCode := 0
+		err = session.cmd.Wait()
+		if err != nil {
+			if exErr, ok := err.(*exec.ExitError); !ok {
+				t.Logf("failed to coerce exit error: %v", err)
+				exitCode = -1
+			} else {
+				exitCode = exErr.ExitCode()
+			}
+		}
+
+		sessCh <- SessInfo{Stdout: string(stdout), Stderr: string(stderr), ExitCode: exitCode}
```

Comment:
```suggestion
		exitCode := 0
		err = session.cmd.Wait()
		if err != nil {
			if exErr, ok := err.(*exec.ExitError); !ok {
				t.Logf("failed to coerce exit error: %v", err)
				exitCode = -1
			} else {
				exitCode = exErr.ExitCode()
			}
		}

		sessCh <- SessInfo{Stdout: session.Stdout.String(), Stderr: session.Stderr.String(), ExitCode: exitCode}
```

To get the contents of `session.Stdout` you don't have to do `stdout, err := io.ReadAll(session.Stdout)` you can simply do `session.Stdout.String()`

Reply:
So I had previously tried out exactly what you said (to try to print it in build job), it doesn't work because we need to do the read before the `wait()`, otherwise content is lost since pipe is closed after wait. Reference: [pipe source code](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/os/exec/exec.go;l=1060-1064) and [here](https://github.com/kubernetes/minikube/blob/2d8bc3e0565733559bc78c0f72e8ab12c046e7e8/test/integration/helpers_test.go#L135) (i.e., session.Stderr = cmd.StderrPipe()).

Reply:
I see, my concern isn't necessarily where we read the contents, just that instead of:
```
		stderr, err := io.ReadAll(session.Stderr)
		if err != nil {
			t.Logf("Failed to read stderr: %v", err)
		}
		stdout, err := io.ReadAll(session.Stdout)
		if err != nil {
			t.Logf("Failed to read stdout: %v", err)
		}
```

It could be replaced with:
```
stderr := session.Stderr.String()
stdout := session.Stdout.String()
```

Reply:
Ahh okay. But I don't see a method called `String()` for `*bufio.Reader` (which is the type of `Stderr`/`Stdout`). 
Got the following compile time error: `session.Stderr.String undefined (type *bufio.Reader has no field or method String)`

Let me know if I'm missing something!